src/index.js

```diff
@@ -24,16 +24,19 @@ const transform = (successFn: Function, failFn: Function, input: Array<any>): an
 /**
  *
  * @param {Function} predicate validation function to apply inputs on
- * @param {String} errorMsg error message to return in case of fail
+ * @param {String|Function} errorMsg error message to return in case of fail
  * @param {*} value the actual value
  * @param {Object} inputs the input object - in case the predicate function needs access to dependent values
  * @returns {Boolean}
  */
 const runPredicate = ([predicate, errorMsg]:[Function, string],
   value:any,
-  inputs:Object) => predicate(value, inputs)
-  ? true
-  : errorMsg
+  inputs:Object) => {
+  if (predicate(value, inputs)) {
```

Comment:
Maybe we should disable `no-nested-ternary` rule? 
I think we can write this in a readable manner anyway.
`
() => predicate(value, inputs)
  ? true
  : typeof errorMsg === 'function' 
  ? errorMsg(value) 
  : errorMsg
`
Any thoughts on this?